src/bindings/python/src/openvino/frontend/tensorflow/utils.py

```diff
@@ -0,0 +1,204 @@
+# Copyright (C) 2018-2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+# flake8: noqa
+# mypy: ignore-errors
+
+
+from openvino.tools.mo.moc_frontend.shape_utils import get_static_shape
+from openvino.tools.mo.utils.versions_checker import get_environment_setup  # pylint: disable=no-name-in-module
+from openvino.tools.mo.utils.error import Error
+from distutils.version import LooseVersion
+import logging as log
+
+
+def trace_tf_model_if_needed(input_model, placeholder_shapes, placeholder_data_types, example_input):
+    import tensorflow as tf
+    if not isinstance(input_model, (tf.keras.layers.Layer, tf.Module, tf.keras.Model, tf.types.experimental.GenericFunction)):
+        return input_model
+    return trace_tf_model(input_model, placeholder_shapes, placeholder_data_types, example_input)
+
+
+def get_input_spec_from_model(model):
+    import tensorflow as tf
+    if hasattr(model, "_build_input_shape") and model._build_input_shape is not None:
+        if isinstance(model._build_input_shape, list):
+            input_spec = [[tf.TensorSpec(shape) for shape in model._build_input_shape]]
+        else:
+            input_spec = [tf.TensorSpec(model._build_input_shape)]
+    else:
+        input_spec = [tf.TensorSpec(None)]
+    return input_spec
+
+
+def create_example_input_by_user_shapes(input_shapes, input_types):
+    import tensorflow as tf
+    if input_shapes is None:
+        return None
+    if isinstance(input_shapes, dict):
+        res = {}
+        for name, shape in input_shapes.items():
+            shape = get_static_shape(shape, 1)
+            args = {}
+            if name in input_types:
+                args["dtype"] = input_types[name]
+            tensor = tf.random.uniform(shape=shape, **args)
+            res[name] = tensor
+        return res
+    elif isinstance(input_shapes, list):
+        res = []
+        for idx, shape in enumerate(input_shapes):
+            shape = get_static_shape(shape, 1)
+            args = {}
+            if idx < len(input_types):
+                args["dtype"] = input_types[idx]
+            tensor = tf.random.uniform(shape=shape, **args)
+            res.append(tensor)
+        return res
+    raise Error("Could not create example input by provided shape {}".format(input_shapes))
+
+
```

Comment:
please add a description for this function and its arguments. `input_needs_packing` is not quite clear. And How to determine `use_example_input`?